osu.Game/Screens/Edit/EditorBeatmap.cs

```diff
@@ -201,6 +202,25 @@ public void RemoveAt(int index)
             updateHitObject(null, true);
         }
 
+        /// <summary>
+        /// Clears all <see cref="HitObjects"/> from this <see cref="EditorBeatmap"/>.
+        /// </summary>
+        public void Clear()
+        {
+            var removed = HitObjects.ToList();
+
+            mutableHitObjects.Clear();
+
+            foreach (var b in startTimeBindables)
+                b.Value.UnbindAll();
+            startTimeBindables.Clear();
+
+            foreach (var h in removed)
+                HitObjectRemoved?.Invoke(h);
+
+            updateHitObject(null, true);
```

Comment:
So as a general note, not necessarily relevant to this PR in itself - I haven't read a lot of editor code, but when reading this clear implementation it feels like the code could use some restructuring - a lot of things are done here. I was especially confused as to why `SelectedHitObjects` isn't being cleared here but found out that it does get cleared - and it gets cleared because the hit objects remove *themselves* from that list via `HitObjectRemoved` which is not something I would have expected.

Reply:
The editor's probably going to go through a dozen more refactors before anything really starts to feel good, I think. It's a complex system of interactions, and this was the result of a refactoring of selections not too long ago.